extensions/shared/external-media/sources/with-media.js

```diff
@@ -161,8 +162,13 @@ export default function withMedia() {
 			};
 
 			copyMedia = ( items, apiUrl ) => {
+				const { addToGallery, multiple, noticeOperations, onClose, onSelect, value } = this.props;
+				const galleryImages = addToGallery
+					? value.map( id => select( 'core' ).getMedia( Number( id ) ) )
+					: [];
+
```

Comment:
With select approach

```suggestion
				const { addToGallery, multiple, noticeOperations, onClose, onSelect, value, getMediaById } = this.props;
				const galleryImages = addToGallery
					? value.map( getMediaById )
					: [];

```

Reply:
Also, you need to compose this component:


```es6
		return compose( [
			withNotices,
			withSelect( ( select ) => {
				const { getMedia } = select( 'core' );
				return {
					getMediaById: ( id ) => getMedia( Number( id ) ),
				};
			} ),
		] )( WithMediaComponent );
```

Reply:
Yeah, I was going back and forth on that when I worked on it.

What's the benefit of `withSelect` over `select`?

Reply:
I'm not totally about the difference in detail and in terms of performance and/or functionality, but the fact it applies a composition with HOC is enough to recommend it IMO. It keeps divided the component with the access data in a clearer way.

It allows the component doesn't worry about the data and how to get it but just picking them up from one of its properties.